provider/dummy/environs.go

```diff
@@ -1797,3 +1799,44 @@ func (*environ) SSHAddresses(ctx context.ProviderCallContext, addresses []networ
 func (*environ) SuperSubnets(ctx context.ProviderCallContext) ([]string, error) {
 	return nil, errors.NotSupportedf("super subnets")
 }
+
+// SetAgentStatus sets the presence for a particular agent in the fake presence implementation.
+func (e *environ) SetAgentStatus(agent string, status presence.Status) {
+	estate, err := e.state()
+	if err != nil {
+		panic(err)
+	}
+	estate.presence.agent[agent] = status
+}
+
+// fakePresence returns alive for all agent alive requests.
+type fakePresence struct {
```

Comment:
Can we just embed the Recorder interface here to not have to provide empty implementations of everything? Are all those stub methods called?